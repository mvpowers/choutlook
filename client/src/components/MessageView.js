import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Reply } from './';

const Container = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  background-color: #f8f8f8;
`;

const Message = styled.div`
  background-color: #fff;
  border: 1px solid #eaeaea;
  margin: 10px;
`;

const Title = styled.div`
  padding: 20px;
`;

const Sender = styled.div`
  display: flex;
  flex-direction: row;
  padding: 10px;
  font-size: 12px;
  align-items: center;
`;

const Bubble = styled.div`
  display: inline-block;
  border-radius: 100px;
  background-color: #1e7145;
  color: #fff;
  padding: 10px 10px;
  margin: 10px;
`;

const Text = styled.div`
  padding: 20px;
  font-size: 14px;
`;

const MessageView = ({
  sendMsg,
  username,
  editingUsername,
  handleChange,
  submitSendMsg,
  discardSendMsg,
  submitOnEnter,
  handleFocus,
  handleBlur,
  toggleEditUsername,
}) => (
  <Container>
    <Title>Re: Super Important Business</Title>
    <Message>
      <Sender>
        <Bubble>CH</Bubble>
        admin@choutlook.com
      </Sender>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet
        consequuntur culpa cupiditate delectus est explicabo fugit harum illum
        incidunt ipsam, minima neque nostrum odit officia placeat possimus
      </Text>
    </Message>
    <Reply
      sendMsg={sendMsg}
      username={username}
      editingUsername={editingUsername}
      toggleEditUsername={toggleEditUsername}
      handleChange={handleChange}
      submitSendMsg={submitSendMsg}
      discardSendMsg={discardSendMsg}
      submitOnEnter={submitOnEnter}
      handleFocus={handleFocus}
      handleBlur={handleBlur}
    />
  </Container>
);

MessageView.propTypes = {
  sendMsg: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  editingUsername: PropTypes.bool.isRequired,
  toggleEditUsername: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  submitSendMsg: PropTypes.func.isRequired,
  discardSendMsg: PropTypes.func.isRequired,
  submitOnEnter: PropTypes.func.isRequired,
  handleFocus: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
};

export default MessageView;
