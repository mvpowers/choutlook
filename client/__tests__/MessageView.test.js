import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MessageView from '../src/components/MessageView';

Enzyme.configure({ adapter: new Adapter() });

describe('MessageView', () => {
  const toggleEditUsername = jest.fn();
  const handleChange = jest.fn();
  const submitSendMsg = jest.fn();
  const discardSendMsg = jest.fn();
  const submitOnEnter = jest.fn();
  const handleFocus = jest.fn();
  const handleBlur = jest.fn();
  it('should be defined', () => {
    expect(MessageView).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <MessageView
        sendMsg="sendMsg"
        username="username"
        editingUsername={false}
        toggleEditUsername={toggleEditUsername}
        handleChange={handleChange}
        submitSendMsg={submitSendMsg}
        discardSendMsg={discardSendMsg}
        submitOnEnter={submitOnEnter}
        handleFocus={handleFocus}
        handleBlur={handleBlur}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
