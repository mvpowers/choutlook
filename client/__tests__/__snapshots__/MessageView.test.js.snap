// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageView should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageView
    discardSendMsg={[MockFunction]}
    editingUsername={false}
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    sendMsg="sendMsg"
    submitOnEnter={[MockFunction]}
    submitSendMsg={[MockFunction]}
    toggleEditUsername={[MockFunction]}
    username="username"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MessageView__Title>
          Re: Super Important Business
        </MessageView__Title>,
        <MessageView__Message>
          <MessageView__Sender>
            <MessageView__Bubble>
              CH
            </MessageView__Bubble>
            admin@choutlook.com
          </MessageView__Sender>
          <MessageView__Text>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus
          </MessageView__Text>
        </MessageView__Message>,
        <Reply
          discardSendMsg={[MockFunction]}
          editingUsername={false}
          handleBlur={[MockFunction]}
          handleChange={[MockFunction]}
          handleFocus={[MockFunction]}
          sendMsg="sendMsg"
          submitOnEnter={[MockFunction]}
          submitSendMsg={[MockFunction]}
          toggleEditUsername={[MockFunction]}
          username="username"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Re: Super Important Business",
        },
        "ref": null,
        "rendered": "Re: Super Important Business",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MessageView__Sender>
              <MessageView__Bubble>
                CH
              </MessageView__Bubble>
              admin@choutlook.com
            </MessageView__Sender>,
            <MessageView__Text>
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus
            </MessageView__Text>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <MessageView__Bubble>
                  CH
                </MessageView__Bubble>,
                "admin@choutlook.com",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "CH",
                },
                "ref": null,
                "rendered": "CH",
                "type": [Function],
              },
              "admin@choutlook.com",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus",
            },
            "ref": null,
            "rendered": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "discardSendMsg": [MockFunction],
          "editingUsername": false,
          "handleBlur": [MockFunction],
          "handleChange": [MockFunction],
          "handleFocus": [MockFunction],
          "sendMsg": "sendMsg",
          "submitOnEnter": [MockFunction],
          "submitSendMsg": [MockFunction],
          "toggleEditUsername": [MockFunction],
          "username": "username",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MessageView__Title>
            Re: Super Important Business
          </MessageView__Title>,
          <MessageView__Message>
            <MessageView__Sender>
              <MessageView__Bubble>
                CH
              </MessageView__Bubble>
              admin@choutlook.com
            </MessageView__Sender>
            <MessageView__Text>
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus
            </MessageView__Text>
          </MessageView__Message>,
          <Reply
            discardSendMsg={[MockFunction]}
            editingUsername={false}
            handleBlur={[MockFunction]}
            handleChange={[MockFunction]}
            handleFocus={[MockFunction]}
            sendMsg="sendMsg"
            submitOnEnter={[MockFunction]}
            submitSendMsg={[MockFunction]}
            toggleEditUsername={[MockFunction]}
            username="username"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Re: Super Important Business",
          },
          "ref": null,
          "rendered": "Re: Super Important Business",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MessageView__Sender>
                <MessageView__Bubble>
                  CH
                </MessageView__Bubble>
                admin@choutlook.com
              </MessageView__Sender>,
              <MessageView__Text>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus
              </MessageView__Text>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <MessageView__Bubble>
                    CH
                  </MessageView__Bubble>,
                  "admin@choutlook.com",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "CH",
                  },
                  "ref": null,
                  "rendered": "CH",
                  "type": [Function],
                },
                "admin@choutlook.com",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus",
              },
              "ref": null,
              "rendered": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. A ab amet consequuntur culpa cupiditate delectus est explicabo fugit harum illum incidunt ipsam, minima neque nostrum odit officia placeat possimus",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "discardSendMsg": [MockFunction],
            "editingUsername": false,
            "handleBlur": [MockFunction],
            "handleChange": [MockFunction],
            "handleFocus": [MockFunction],
            "sendMsg": "sendMsg",
            "submitOnEnter": [MockFunction],
            "submitSendMsg": [MockFunction],
            "toggleEditUsername": [MockFunction],
            "username": "username",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
